pkg/controller/certificates/cleaner/cleaner.go

```diff
@@ -91,20 +92,21 @@ func (ccc *CSRCleanerController) Run(ctx context.Context, workers int) {
 
 // worker runs a thread that dequeues CSRs, handles them, and marks them done.
 func (ccc *CSRCleanerController) worker(ctx context.Context) {
+	logger := klog.FromContext(ctx)
 	csrs, err := ccc.csrLister.List(labels.Everything())
 	if err != nil {
-		klog.Errorf("Unable to list CSRs: %v", err)
+		logger.Error(err, "Unable to list CSRs")
 		return
 	}
 	for _, csr := range csrs {
 		if err := ccc.handle(ctx, csr); err != nil {
-			klog.Errorf("Error while attempting to clean CSR %q: %v", csr.Name, err)
+			logger.Error(err, "Error while attempting to clean CSR", "csr", csr.Name)
 		}
 	}
 }
 
 func (ccc *CSRCleanerController) handle(ctx context.Context, csr *capi.CertificateSigningRequest) error {
-	if isIssuedPastDeadline(csr) || isDeniedPastDeadline(csr) || isFailedPastDeadline(csr) || isPendingPastDeadline(csr) || isIssuedExpired(csr) {
+	if isIssuedPastDeadline(ctx, csr) || isDeniedPastDeadline(ctx, csr) || isFailedPastDeadline(ctx, csr) || isPendingPastDeadline(ctx, csr) || isIssuedExpired(ctx, csr) {
```

Comment:
This functions are so simple that they are unlikely to ever need a context. Better pass a logger.